src/Build.UnitTests/FixPathOnUnix_Tests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Xunit;
+
+namespace Microsoft.Build.UnitTests
+{
+    [PlatformSpecific(Xunit.PlatformID.AnyUnix)]
+    public  class FixPathOnUnixTests
+    {
+        [Fact]
+        public void TestPathFixupInMetadata()
+        {
+            string buildProjectContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003' >
+                    <Target Name='Build'>
+                        <MSBuild Projects='projectDirectory/main.proj' />
+                    </Target>
+               </Project>";
+
+            string mainProjectContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003' >
+                    <UsingTask TaskName='LogTaskPropertiesTask' AssemblyName='Microsoft.Build.Engine.UnitTests' />
+                    <ItemGroup>
+                        <Item0 Include='xyz'>
+                            <Md0>lib\foo.dll</Md0>
+                        </Item0>
+                    </ItemGroup>
+                    <Target Name='Build'>
+                        <LogTaskPropertiesTask Items='@(Item0)' />
+                    </Target>
+                </Project>";
+
+            string buildProjectPath = ObjectModelHelpers.CreateFileInTempProjectDirectory("build.proj", buildProjectContents);
+            string mainProjectPath = ObjectModelHelpers.CreateFileInTempProjectDirectory("projectDirectory/main.proj", mainProjectContents);
+            ObjectModelHelpers.CreateFileInTempProjectDirectory(Path.Combine("projectDirectory", "lib", "foo.dll"), "just a text file");
+
+            var projColln = new ProjectCollection();
```

Comment:
Also, use the Disposable `Microsoft.Build.UnitTests.Helpers.TestProjectWithFiles` instead of the static method. It automatically clears the created files in the Dispose.